src/slider/internal.tsx

```diff
@@ -0,0 +1,188 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useState, useRef } from 'react';
+import clsx from 'clsx';
+
+import styles from './styles.css.js';
+import { getBaseProps } from '../internal/base-component/index.js';
+import { SliderProps } from './interfaces.js';
+import useBaseComponent from '../internal/hooks/use-base-component/index.js';
+import { fireNonCancelableEvent } from '../internal/events/index.js';
+import { useFormFieldContext } from '../internal/context/form-field-context.js';
+import { useUniqueId } from '../internal/hooks/use-unique-id/index.js';
+import Tooltip from '../internal/components/tooltip/index.js';
+import SliderLabels from './slider-labels.js';
+import { getPercent, getStepArray, findLowerAndHigherValues, valuesAreValid, THUMB_SIZE } from './utils.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+
+export default function InternalSlider({
+  value,
+  min,
+  max,
+  onChange,
+  step,
+  disabled,
+  ariaLabel,
+  referenceValues,
+  tickMarks,
+  hideFillLine,
+  valueFormatter,
+  ...rest
+}: SliderProps) {
+  const { __internalRootRef } = useBaseComponent('Slider');
+  const baseProps = getBaseProps(rest);
+
+  const handleRef = useRef<HTMLDivElement>(null);
+  const [showTooltip, setShowTooltip] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+  const labelsId = useUniqueId('labels');
+  const hasValue = value !== undefined;
+  const formFieldContext = useFormFieldContext(rest);
+  const { ariaLabelledby, ariaDescribedby, controlId, invalid } = formFieldContext ?? rest;
```

Comment:
Can formFieldContext be undefined?

Reply:
I assumed it was undefined when a Slider was used outside of a form field. So, use the form field properties when inside a form field, and use from props when not.

Reply:
The context is always defined but when used outside of the form field the individual properties will all be default: undefined or false.